src/synapse.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper module used to manage interactions with Synapse."""
+
+import logging
+import typing
+
+import ops
+from ops.pebble import Check, ExecError
+
+from charm_state import CharmState
+from charm_types import ExecResult
+from constants import CHECK_READY_NAME, COMMAND_MIGRATE_CONFIG, COMMAND_PATH, SYNAPSE_PORT
+from exceptions import CommandMigrateConfigError
+
+logger = logging.getLogger(__name__)
+
+
+class Synapse:  # pylint: disable=too-few-public-methods
```

Comment:
Do you really need to disable the too-few-public-methods ? There are 3 public methods and the default in pylint is 2

Reply:
Good catch :) Removed.